server/manager/src/main/java/org/apache/accumulo/manager/FateServiceHandler.java

```diff
@@ -938,15 +945,18 @@ public Path mkTempDir(long opid) throws IOException {
   }
 
   @Override
-  public boolean cancelFateOperation(TInfo tinfo, TCredentials credentials, long opid)
+  public boolean cancelFateOperation(TInfo tinfo, TCredentials credentials, TFateId opid)
       throws ThriftSecurityException, ThriftNotActiveServiceException {
 
     if (!manager.security.canPerformSystemActions(credentials)) {
       throw new ThriftSecurityException(credentials.getPrincipal(),
           SecurityErrorCode.PERMISSION_DENIED);
     }
 
-    return manager.fate().cancel(opid);
+    return manager.fate(getType(opid)).cancel(opid.getTid());
   }
 
+  private static FateInstanceType getType(TFateId opid) {
+    return FateInstanceType.valueOf(opid.getType().name());
```

Comment:
```suggestion
    return FateInstanceType.fromThrift(opid.getType());
```

could add the following the the FateInstanceType enum and call as above.  The impl below uses switch and avoids the string valueOf lookup, may be a bit more efficient.

```java
  public static FateInstanceType fromThrift(TFateInstanceType tfit) {
    switch (tfit) {
      case USER:
        return FateInstanceType.USER;
      case META:
        return FateInstanceType.META;
      default:
        throw new IllegalStateException("Unknown type "+tfit);
    }
  }
```